src/c/device-can.c

```diff
@@ -169,11 +150,27 @@ static bool can_get_handler(void *impl, const devsdk_device_t *device,
 	struct can_frame frame;
 	uint32_t data[128] = {0};
 	int nbytes = 0;
+	int iret = -1;
+
+	end_dev_params *end_dev_params_ptr = (end_dev_params *)device->address;
 
 	if (device == NULL) {
 		return successful_get_request;
 	}
 
+	if(!end_dev_params_ptr->can_IsOpened)
+	{
+		iret = OpenCan(impl, device); 
+		if(iret != 0)
+			return successful_get_request;
```

Comment:
the mutex should be unlocked before the `return` statement

Reply:
Modified the code